include/pybind11/detail/class.h

```diff
@@ -581,7 +581,15 @@ extern "C" inline int pybind11_getbuffer(PyObject *obj, Py_buffer *view, int fla
         return -1;
     }
     std::memset(view, 0, sizeof(Py_buffer));
-    buffer_info *info = tinfo->get_buffer(obj, tinfo->get_buffer_data);
+    buffer_info *info = nullptr;
+    try {
+        info = tinfo->get_buffer(obj, tinfo->get_buffer_data);
+    } catch (...) {
+        try_translate_exceptions();
```

Comment:
indeed, applied

Reply:
Just from inspecting [the sources](https://github.com/pybind/pybind11/blob/66c3774a6402224b1724329c81c880e76633a92b/include/pybind11/pybind11.h#L1740), I'm thinking `info` could be `nullptr` here, which would lead to segfaults further down.

Does that make sense to you?

If true, it is an existing problem, but I'd lean towards fixing it in this PR while we're at it. Simply (untested):

```
if (info == nullptr) {
    pybind11_fail("FATAL UNEXPECTED SITUATION: tinfo->get_buffer() returned nullptr.");
}
```

Reply:
this would also crash the app immediately... but I guess it's better to at least be explicit about the reason, so added

Reply:
Thanks, yes, much better. I'm used to looking a traceback hundreds of levels deep. Seeing or not seeing that exception can make the difference between an immediate aha and several days lost.